scripts/lighthouse/index.js

```diff
@@ -0,0 +1,40 @@
+const stoplight = res => (res >= 90 ? '🟢' : res >= 75 ? '🟠' : '🔴');
+const normalizeScore = res => Math.round(res * 100);
+const formatScore = res => {
+  const normalizedScore = normalizeScore(res);
+  return `${stoplight(normalizedScore)} ${normalizedScore}`;
+};
+
+/**
+ * `core` is in scope from https://github.com/actions/github-script
+ * This is a CJS formatted file because using ESM here requires setting `type: module` in the package.json and that broke out site capabilities
+ */
+module.exports = ({ core }) => {
+  // this will be the shape of https://github.com/treosh/lighthouse-ci-action#manifest
+  const results = JSON.parse(process.env.LIGHTHOUSE_RESULT);
+
+  // this will be the shape of https://github.com/treosh/lighthouse-ci-action#links
+  const links = JSON.parse(process.env.LIGHTHOUSE_LINKS);
+
+  // start creating our markdown table
+  const header = [
+    'Lighthouse Results',
+    'URL | Performance | Accessibility | Best Practices | SEO | Report',
+    '| - | - | - | - | - | - |',
+  ];
+
+  // map over each url result, formatting and linking to the output
+  const urlResults = results.map(({ url, summary }) => {
+    return `[${url.replace(
```

Comment:
Could you break all these long variables into constants? Just to make it easier reading this string template? It'd be great if you could also add some sort of unit test here :)

Reply:
done via https://github.com/nodejs/nodejs.org/pull/6047/commits/41f9cd3731289f5348e74bac814e6bef05a4e96e